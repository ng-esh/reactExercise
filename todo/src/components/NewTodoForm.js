import React, { useState } from "react";
import "../styles/NewTodoForm.scss";

function NewTodoForm({ addTodo }) {
    const [task, setTask] = useState("");
  
    const handleSubmit = (e) => {
      e.preventDefault();
      if (task.trim() === "") return;
      addTodo(task);
      setTask("");
    };
  
    return (
      <form className="new-todo-form" onSubmit={handleSubmit}>
        <input
          type="text"
          value={task}
          onChange={(e) => setTask(e.target.value)}
          placeholder="Enter a new todo"
          data-testid="todo-input"
        />
        <button type="submit" data-testid="add-todo-btn">Add Todo</button>
      </form>
    );
  
}

export default NewTodoForm;