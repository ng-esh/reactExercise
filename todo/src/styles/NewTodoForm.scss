.new-todo-form {
  display: flex;
  gap: 10px;
  margin-bottom: 20px;

  input {
    flex-grow: 1;
    padding: 5px;
    border: 1px solid #ccc;
    border-radius: 4px;
  }

  button {
    padding: 5px 10px;
    background: #007bff;
    color: white;
    border: none;
    cursor: pointer;
    border-radius: 4px;
    transition: background 0.3s;

    &:hover {
      background: #0056b3;
    }
  }
}
