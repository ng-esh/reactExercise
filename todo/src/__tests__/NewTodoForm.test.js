import { render, fireEvent, screen } from "@testing-library/react";
import NewTodoForm from "../components/NewTodoForm.js";

test("renders without crashing", () => {
  render(<NewTodoForm addTodo={() => {}} />);
});

test("matches snapshot", () => {
  const { asFragment } = render(<NewTodoForm addTodo={() => {}} />);
  expect(asFragment()).toMatchSnapshot();
});

test("allows user to enter a new todo", () => {
  render(<NewTodoForm addTodo={() => {}} />);
  const input = screen.getByTestId("todo-input");
  fireEvent.change(input, { target: { value: "New Task" } });
  expect(input.value).toBe("New Task");
});