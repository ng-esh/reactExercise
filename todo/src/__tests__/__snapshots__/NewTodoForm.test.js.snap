// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="new-todo-form"
  >
    <input
      data-testid="todo-input"
      placeholder="Enter a new todo"
      type="text"
      value=""
    />
    <button
      data-testid="add-todo-btn"
      type="submit"
    >
      Add Todo
    </button>
  </form>
</DocumentFragment>
`;
